### PR TITLE
feat(shared-data): Convert GEN1/GEN2 pipettes to the new shared-data schema

### PR DESCRIPTION
**Pull request by Laura-Danielle**
# Overview

Closes #RLIQ 250. Check out [this confluence doc](https://opentrons.atlassian.net/wiki/spaces/RPDO/pages/edit-v2/3589931027?draftShareId=4150801a-9195-4efb-bfe0-18807ae3cc40&inEditorTemplatesPanel=auto_closed) for further information on how to use the script for building new pipette models.

# Changelog

- Update shared data definitions for GEN1/GEN2 pipettes
- Add in a script to automate creating new pipette configuration definitions

# Review requests

Double check the script. Let me know if anything doesn't make sense.

# Risk assessment

Low. Not being currently used.

### CODE REVIEW

File: shared-data/pipette/definitions/2/geometry/eight_channel/p300/1_0.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "nozzleOffset": [0.0, 31.5, 0.8],
+  "pathTo3D": "/Users/lauracox/Desktop/Work/opentrons/shared-data/pipette/definitions/2/geometry/eight_channel/p300/placeholder.gltf"
```
Comment (sfoster1) [inline shared-data/pipette/definitions/2/geometry/eight_channel/p300/1_0.json]: these paths need to be sanitized - should be relative to file i assume

File: shared-data/python/opentrons_shared_data/pipette/build_json_script.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
+import argparse
```
Comment (sfoster1) [inline shared-data/python/opentrons_shared_data/pipette/build_json_script.py]: for consistency with scripts in other places we should put this in a `scripts` submodule or something

File: shared-data/python/opentrons_shared_data/pipette/build_json_script.py [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+        )
+
+        quirks_list[model] = model_configurations["quirks"]
+        split_version = full_version.split(".")
+        file_name = (
+            f"{split_version[0]}_{split_version[1] if len(split_version) > 1 else 0}"
+        )
+        current_pipette_path = Path(physical_model.channels.name.lower()) / pipette_type
+
+        path_to_3d = GEOMETRY_ROOT / current_pipette_path / "placeholder.gltf"
```
Comment (sfoster1) [inline shared-data/python/opentrons_shared_data/pipette/build_json_script.py]: probably the right place to put the path changes

**Review by sfoster1 — CHANGES_REQUESTED** [review]
- `pathTo3D` entries in geometry configs need to be sanitized, either to relative paths or to "json anchor paths" e.g. absolute paths relative to the data root that start with `#`
- make sure we mark the gltf files binary in git

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Oh also if we don't have a test that just schema checks everything we should add one

**Comment by Laura-Danielle** [comment]
> Oh also if we don't have a test that just schema checks everything we should add one

The schema checks happen in js at `shared-data/js/__tests__/pipetteSchemaV2.test.ts` which also happen to be failing cus I missed something from the schema 😉

**Review by sfoster1 — APPROVED** [review]
Looks good to me generally, though I think maybe we want a `#` anchor in the gltf paths? could go either way

**Pull request merged**